README.md

```diff
@@ -292,6 +306,59 @@ hayhooks pipeline run <pipeline_name> --dir files_to_index --file file1.pdf --fi
 hayhooks pipeline run <pipeline_name> --file file.pdf --param 'question="is this recipe vegan?"'
 ```
 
+## MCP support
+
+**NOTE: You'll need to run at least Python 3.10+ to use the MCP server.**
+
+### MCP Server
+
+Hayhooks now supports the [Model Context Protocol](https://modelcontextprotocol.io/) and can act as a [MCP Server](https://modelcontextprotocol.io/docs/concepts/architecture).
+
+It will automatically expose the deployed pipelines as [MCP Tools](https://modelcontextprotocol.io/docs/concepts/tools), using [Server-Sent Events (SSE)](https://modelcontextprotocol.io/docs/concepts/transports#server-sent-events-sse) as MCP Transport.
+
+### Create a PipelineWrapper for exposing a Haystack pipeline as a MCP Tool
+
+A [MCP Tool](https://modelcontextprotocol.io/docs/concepts/tools) requires the following properties:
+
+- `name`: The name of the tool.
+- `description`: The description of the tool.
+- `inputSchema`: A JSON Schema object describing the tool's input parameters.
+
+For each deployed pipeline, Hayhooks will:
+
+- Use the pipeline wrapper `name` as MCP Tool `name`.
+- Use the pipeline wrapper **`run_api` method docstring** as MCP Tool `description`.
+- Generate a Pydantic model from the `inputSchema` using the **`run_api` method arguments as fields**.
+
+Here's an example of a PipelineWrapper implementation for the `chat_with_website` pipeline which can be used as a MCP Tool:
+
+```python
+from pathlib import Path
+from typing import List
+from haystack import Pipeline
+from hayhooks import BasePipelineWrapper
+
+
+class PipelineWrapper(BasePipelineWrapper):
+    def setup(self) -> None:
+        pipeline_yaml = (Path(__file__).parent / "chat_with_website.yml").read_text()
+        self.pipeline = Pipeline.loads(pipeline_yaml)
+
+    def run_api(self, urls: List[str], question: str) -> str:
+        #
+        # NOTE: The following docstring will be used as MCP Tool description
+        #
+        """
+        Ask a question about one or more websites using a Haystack pipeline.
+        """
+        result = self.pipeline.run({"fetcher": {"urls": urls}, "prompt": {"query": question}})
+        return result["llm"]["replies"][0]
+```
+
+Note that if you omit the docstring, the pipeline will be deployed but **will not be listed as a MCP Tool** (you will see a warning in the Hayhooks logs).
```

Comment:
Is this still true? I am not sure.

In any case, let's mention here in the README how to opt-out from MCP (using `skip_mcp = True`)

Reply:
@anakin87 sure, I've just update README adding also a section about how to start the MCP server (was missing too).